dev/InfoBadge/InfoBadgeImpl/InfoBadgeImpl.cpp

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include "InfoBadgeImpl.h"
+
+namespace InfoBadgeImpl
+{
+#pragma region ControlOverrides
+    winrt::Size InfoBadgeImpl::MeasureOverrideImpl(Size const& defaultDesiredSize)
+    {
+        if (defaultDesiredSize.Width < defaultDesiredSize.Height)
+        {
+            return { defaultDesiredSize.Height, defaultDesiredSize.Height };
+        }
+        return defaultDesiredSize;
+    }
+#pragma endregion 
+
+#pragma region OnValuePropertyChanged
+    void InfoBadgeImpl::ValidateValuePropertyImpl(int value)
+    {
+        if (value < -1)
+        {
+            throw winrt::hresult_out_of_bounds(L"Value must be equal to or greater than -1");
+        }
+    }
+#pragma endregion 
+
+#pragma region GoToAppropriateDisplayKindState
+    InfoBadgeDisplayKindStates InfoBadgeImpl::CalculateAppropriateDisplayKindStateImpl(int value, bool iconExists, bool iconIsFontIcon)
+    {
+        if (value >= 0)
+        {
+            return InfoBadgeDisplayKindStates::Value;
+        }
+
+        return CalculateIconOrDotDisplayKindState(iconExists, iconIsFontIcon);
+    }
+
+    namespace
+    {
+        InfoBadgeDisplayKindStates CalculateIconOrDotDisplayKindState(bool iconExists, bool iconIsFontIcon)
```

Comment:
Are these method necessary ? Would it be more readable to just have a bigger if/else if or case statement in above method instead ? Or are these separate methods so you can test them separately ?